### PR TITLE
Add perf section on reducing runtime overhead 

### PR DESCRIPTION
**Pull request by roji**

### CODE REVIEW

File: entity-framework/core/performance/advanced-performance-topics.md [hunk 1]
```diff
@@ -109,3 +109,16 @@ Even if the sub-millisecond difference seems small, keep in mind that the consta
...
+
+* Turn on [DbContext pooling](#dbcontext-pooling); our benchmarks show that this feature can have a decisive impact on high-perf, low-latency applications.
+  * Make sure that the `maxPoolSize` corresponds to your usage scenario; if it is too low, DbContext instances will be constantly created and disposed, degrading performance. Setting it too high may needlessly consume memory as unused DbContext instances are maintained in the pool.
+  * For an extra tiny perf boost, consider using <xref:Microsoft.EntityFrameworkCore.Infrastructure.PooledDbContextFactory%601> instead of having DI inject context instances directly (EF Core 6 and above). DI management of DbContext pooling incurs a slight overhead.
+* Use compiled queries for hot queries.
```
Comment (AndriySvyryd) [inline entity-framework/core/performance/advanced-performance-topics.md:120]: "precompiled"?

File: entity-framework/core/performance/advanced-performance-topics.md [hunk 2]
```diff
@@ -109,3 +109,16 @@ Even if the sub-millisecond difference seems small, keep in mind that the consta
...
+
+* Turn on [DbContext pooling](#dbcontext-pooling); our benchmarks show that this feature can have a decisive impact on high-perf, low-latency applications.
+  * Make sure that the `maxPoolSize` corresponds to your usage scenario; if it is too low, DbContext instances will be constantly created and disposed, degrading performance. Setting it too high may needlessly consume memory as unused DbContext instances are maintained in the pool.
+  * For an extra tiny perf boost, consider using <xref:Microsoft.EntityFrameworkCore.Infrastructure.PooledDbContextFactory%601> instead of having DI inject context instances directly (EF Core 6 and above). DI management of DbContext pooling incurs a slight overhead.
```
Comment (AndriySvyryd) [inline entity-framework/core/performance/advanced-performance-topics.md:119]: Docs for `Microsoft.EntityFrameworkCore.Infrastructure.PooledDbContextFactory` haven't been published yet

**Review by AndriySvyryd — APPROVED** [review]

File: entity-framework/core/performance/advanced-performance-topics.md (hunk 1 above)
Comment (roji) [reply entity-framework/core/performance/advanced-performance-topics.md:120]: Sure. We don't have documentation for that feature yet, so it has no name :)

Will change to precompiled.

File: entity-framework/core/performance/advanced-performance-topics.md (hunk 2 above)
Comment (roji) [reply entity-framework/core/performance/advanced-performance-topics.md:119]: Yeah, same for EnableThreadSafetyChecks. Will remove the links for now and just fence.

**Pull request merged**